Bio_Seq_Analysis_Tool.py

```diff
@@ -0,0 +1,163 @@
+    if quality_threshold > 40:
+        raise ValueError(f'Wrong quality threshold!')
+    analysed_seq = {}
+    error_seq = {}
+    for seq in seqs.items():
+        if fq.analyse_gc(seq[1][0]) >= gc_bounds[0] and fq.analyse_gc(seq[1][0]) <= gc_bounds[1]:
+            if fq.analyse_length(seq[1][0]) >= length_bounds[0] and fq.analyse_length(seq[1][0]) <= length_bounds[1]:
+                if fq.analyse_quality(seq[1][1]) > quality_threshold:
+                    analysed_seq[seq[0]] = (seq[1])
+                else:
+                    error_seq[seq[0]] = (seq[1])
+            else:
+                error_seq[seq[0]] = (seq[1])
+        else:
+            error_seq[seq[0]] = (seq[1])
+    if filtered_file_name == None:
+        new_file_name = file_name 
+    else:
+        new_file_name = filtered_file_name
+    fq.write_fastq(analysed_seq, path_to_file, new_file_name)
+    return error_seq
+
+
+def run_protein_analysis(*args: str) -> Union[List[str], List[float]]:
+    """
+    Launch desired operation with proteins sequences. Pass comma-separated sequences,
+    additional argument (if certain function requires it) and specify function name you want to apply to all
+    sequences.
+    Pass arguments strictly in this order, otherwise it won't be parsed.
+    If the input is a sequence of amino acids written in a three-letter code, then the amino acids must be separated 
+    by a space. If the input is a sequence of amino acids written in a single-letter code, then the amino acids may 
+    not be separated by a space.
+
+    :param args:
+    - seq (str): amino acids sequences for analysis in 1-letter or 3-letter code (as many as you wish)
+    - additional arg (str): necessary parameter for certain functions (for example, specify target protein site)
+    - operation name (str): specify procedure you want to apply
+    :type param args: str
+
+    :return: the result of procedure in list or str format
+    :rtype: Union[List[str], List[float]
+    :raises ValueError: if sequence not protein or sequence in wrong format    
+    """
+    # first value is function name, second is real function, third is number of function arguments
+    function_names = {'change_residues_encoding': [pa.change_residues_encoding, 2],
+                      'is_protein': [pa.is_protein, 1],
+                      'get_seq_characteristic': [pa.get_seq_characteristic, 1],
+                      'find_residue': [pa.find_residue, 2],
+                      'find_site': [pa.find_site, 2],
+                      'calculate_protein_mass': [pa.calculate_protein_mass, 1],
+                      'calculate_average_hydrophobicity': [pa.calculate_average_hydrophobicity, 1],
+                      'get_mrna': [pa.get_mrna, 1],
+                      'calculate_isoelectric_point': [pa.calculate_isoelectric_point, 1],
+                      'analyze_secondary_structure': [pa.analyze_secondary_structure, 1]}
+
+    procedure = args[-1]
+
+    processed_result = []
```

Comment:
```suggestion
    processed_results = []
```
Все таки коллекция. Да и почему не просто `results`?